armi/physics/neutronics/tests/test_crossSectionManager.py

```diff
@@ -119,6 +135,79 @@ def test_createRepresentativeBlock(self):
         self.assertAlmostEqual(avgB.getNumberDensity("U235"), 2.0)
         self.assertEqual(avgB.p.percentBu, 50.0)
 
+        # check that a new block collection of the representative block has right temperatures
+        # this is required for Doppler coefficient calculations
+        newBc = AverageBlockCollection(
+            self.blockList[0].r.blueprints.allNuclidesInProblem
+        )
+        newBc.append(avgB)
+        newBc.calcAvgNuclideTemperatures()
+        self.assertAlmostEqual(newBc.avgNucTemperatures["U235"], 603.0)
+        self.assertAlmostEqual(newBc.avgNucTemperatures["FE56"], 502.0)
+        self.assertAlmostEqual(newBc.avgNucTemperatures["NA23"], 402.0)
+
+    def test_createRepresentativeBlockDissimilar(self):
+
+        # add a different type of block
+        _o, r = test_reactors.loadTestReactor(
+            os.path.join(TEST_ROOT, "detailedAxialExpansion"),
+            inputFileName="armiRun.yaml",
+        )
```

Comment:
Instead of loading up an entire reactor for a single block you can use `loadTestBlock`
https://github.com/terrapower/armi/blob/d341941e570dc731a4b0ab9017acc65c35d43098/armi/reactor/tests/test_blocks.py#L75